src/core/io.jl

```diff
@@ -86,18 +86,18 @@ mean(chain[:mu])      # find the mean of :mu
 mean(chain[:sigma])   # find the mean of :sigma
 ```
 """
-mutable struct Chain <: AbstractChains
+mutable struct Chain{R<:AbstractRange{Int}} <: AbstractChains
```

Comment:
Consider `AbstractRange{<:Integer}`, or perhaps simple `AbstractRange`.

Reply:
Any reason?

Reply:
It's just less concrete, and I don't think we really lose anything. A completely generic `AbstractRange` might be a bit loose though actually.